base/applications/taskmgr/dbgchnl.c

```diff
@@ -243,20 +255,37 @@ static int enum_channel(HANDLE hProcess, EnumChannelCB ce, void* user, unsigned
                         if (!wcscmp(cache[j], buffer + 1)) break;
                     if (j != used_cache) continue;
                     if (used_cache == num_cache)
+                    {
                         cache = HeapReAlloc(GetProcessHeap(), 0, cache, (num_cache *= 2) * sizeof(WCHAR*));
```

Comment:
There is a bug:
If HeapReAlloc fails, it will return NULL, but
> [...] the original memory is not freed, and the original handle and pointer are still valid.

Therefore here in case of failure you are leaking the original "cache" buffer.

Reply:
Indeed, I missed to update this case too.
Fixed.